### ISSUE TITLE
Update Opencast springframework from 3x

### ISSUE THREAD

**Issue by karendolan** [opencast/opencast#4255]
**Describe the bug**
Update Opencast Spring Security, springframework from 3x to supported or replace libraries

**To Reproduce**
This ticket is to track status on the project of updating or replacing the springframework libraries to prevent duplicate effort and encourage collaboration. 

**Environment Information**
All

Reference: https://opencast.org/2022/02/10/crowdfunding-an-update/

**Comment by karendolan** [comment]
Hi All, I have the go ahead to focus on this topic for our site and do not want to duplicate any other efforts. I'll post status here.

**Comment by gregorydlogan** [comment]
ping @lkiesow.  We've talked about this as part of the crowdfunding effors, and Lars has been working on this.  Perhaps we should all chat at some point?

**Comment by karendolan** [comment]
> ping @lkiesow. We've talked about this as part of the crowdfunding effors, and Lars has been working on this. Perhaps we should all chat at some point?

Yes, absolutely! I saw the crowdfunding comments. My plan was to explore and feel the pain directly. Karaf 4.4.1 has upgraded springframework. So I was going to start there.
https://karaf.apache.org/news
https://issues.apache.org/jira/browse/KARAF-7446

**Comment by gregorydlogan** [comment]
IIRC the conclusion was that the amount of pain, especially to maintain feature parity, was not worth it and that alternative solutions would be explored. I believe some kind of auth proxy (as a separate service, focussed exclusively on integrating various auth bits), but don't quote me on that.

**Comment by karendolan** [comment]
> IIRC the conclusion was that the amount of pain, especially to maintain feature parity, was not worth it and that alternative solutions would be explored. I believe some kind of auth proxy (as a separate service, focussed exclusively on integrating various auth bits), but don't quote me on that.

Whoah. @gregorydlogan & @lkiesow, let me know if you have already started down that path. Meanwhile, I'm exploring replacing the `org.springframework.osgi` libs with the comparable and better supported `org.eclipse.gemini.blueprint` libs. Please stop me if you have already tried that. The Gemini Blueprint is better supported now in Karaf because Karak doesn't use its own version of blueprint in the later Karaf versions (per the Karaf release notes)

**Comment by gregorydlogan** [comment]
We spoke about this briefly at the board meeting today.  If you're ok with doing the work, by all means proceed with what you're doing.  It didn't sound like your particular approach was explored, so hopefully it works :D

**Comment by karendolan** [comment]
Hi @gregorydlogan, status is that the gemini.blueprint project was stopped and includes old libraries. I'm not looking at that.
The latest Karaf 4.4.1 has new JAAS based libs that supports Spring 5x, https://karaf.apache.org/manual/latest/security. The vanilla support is not as flexible as blueprint (which replaced the spring-dm which replaced the spring.osgi libs that are currently used in OC). I'm exploring using the new org.apache.karaf.jaas libs in conjunction with the most up to date blueprint support to make a smooth transition from the current mh_default_org style spring config to modern Karaf.

**Comment by lkiesow** [comment]
(Didn't read the thread properly…) 
I just noticed https://github.com/apache/karaf/pull/1522
Seems like Karaf also has Spring.
See also the release notes of the latest 4.2.x release:  https://issues.apache.org/jira/secure/ReleaseNote.jspa?projectId=12311140&version=12351218
I didn't check any further which components are included and which aren't.

**Comment by karendolan** [comment]
Hi @lkiesow, I had a short hiatus, but I am still working on this. I updated a lot of  dependencies and modified the mh_default_org.xml to combines blueprint and spring namespaces. This removes the need for the old, no longer supported, spring-osgi support. I'm currently working on an integration test to sanity check the blueprint spring dependency injections with KarafTestSupport.
Once that is verified, my concern is how to support multi-tenancy with the blueprint deployer in a similar way to how it's being done today with the old defunct OsgiBundleXmlApplicationContext. 
Anyway, I hope to post what I have for discussion in the coming days...

A small example of changes, is adding this karaf features, removing the old springframework from the features, adding karaf spring to the assemblies...
```
+++ b/assemblies/karaf-features/src/main/feature/feature.xml
@@ -16,6 +16,11 @@
     <feature>http</feature>
     <feature>http-whiteboard</feature>
     <feature>scr</feature>
+    <!-- #DCE Karaf 4.4.2+ provides spring and blueprint spring support -->
+    <feature>aries-blueprint-spring</feature>
+    <feature>spring-security</feature>
+    <feature>spring-tx</feature>
+    <feature>spring-web</feature>
     <feature>cxf-jaxrs</feature>
     <feature>cxf-http-jetty</feature>
     ....
```

**Comment by JulianKniephoff** [comment]
Since we have #6991 now, I'll go ahead and take this off https://github.com/orgs/opencast/projects/5 for now.
